app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAboutRobot.kt

```diff
@@ -114,7 +209,77 @@ private fun assertLibrariesUsed() {
         .perform(click())
 
     onView(withId(R.id.action_bar)).check(matches(hasDescendant(withText(containsString("Firefox Preview | OSS Libraries")))))
+    goBackButton().click()
 }
 
 private fun goBackButton() =
     onView(CoreMatchers.allOf(withContentDescription("Navigate up")))
+
+class BuildDateAssertion {
+    companion object {
+        // this pattern represents the following date format: "Monday 12/30 @ 6:49 PM"
+        private const val DATE_PATTERN = "EEEE M/d @ h:m a"
+
+        fun isDisplayedDateAccurate(): ViewAssertion {
+            return ViewAssertion { view, noViewFoundException ->
+                if (noViewFoundException != null) throw noViewFoundException
+
+                val textFromView = (view as TextView).text
+                    ?: throw AssertionError("This view is not of type TextView")
+
+                verifyDateIsWithinRange(textFromView.toString(), 1)
+            }
+        }
+
+        private fun verifyDateIsWithinRange(dateText: String, hours: Int) {
+            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.N_MR1) {
+                val simpleDateFormat = SimpleDateFormat(DATE_PATTERN)
+                val date = simpleDateFormat.parse(dateText)
+                if (!date.isWithinRangeOf(hours)) throw AssertionError("The build date is not within Range.")
+            } else {
+                val textviewDate = getLocalDateTimeFromString(dateText)
+                val buildConfigDate = getLocalDateTimeFromString(BuildConfig.BUILD_DATE)
+
+                if (!buildConfigDate.isEqual(textviewDate) &&
+                    !textviewDate.isWithinRangeOf(hours, buildConfigDate)
+                ) {
+                    throw AssertionError("$textviewDate is not equal to the date within the build config: $buildConfigDate, and are not within a reasonable amount of time from each other.")
+                }
+            }
+        }
+
+        private fun Date.isWithinRangeOf(hours: Int): Boolean {
+            val calendar = Calendar.getInstance()
+            val currentYear = calendar.get(Calendar.YEAR)
+            calendar.time = this
+            calendar.set(Calendar.YEAR, currentYear)
+            val updatedDate = calendar.time
+
+            calendar.add(Calendar.HOUR_OF_DAY, hours)
+            val maxDate = calendar.time
+            calendar.add(
+                Calendar.HOUR_OF_DAY,
+                hours * -2
```

Comment:
this makes sense as you explain it, but it might make sense to add another brief comment as to logic of hour arithmetic